src/Actions/CreateNewUser.php

```diff
@@ -52,15 +54,21 @@ public function create(array $input)
     private function buildValidator(array $input): ValidationValidator
     {
         $rules = [
-            'name'              => ['required', 'string', 'max:255', resolve(PoliteUsername::class)],
+            'name'              => ['required', 'string', 'min:'.Constants::MIN_USERNAME_CHARACTERS, 'max:'.Constants::MAX_USERNAME_CHARACTERS, resolve(PoliteUsername::class)],
             Fortify::username() => $this->usernameRules(),
             'password'          => $this->passwordRules(),
             'terms'             => ['required', 'accepted'],
             'invitation'        => ['sometimes', 'required', 'string'],
         ];
 
         if ($usernameAlt = Config::get('fortify.username_alt')) {
-            $rules[$usernameAlt] = ['required', 'string', 'max:255', 'unique:users', resolve(PoliteUsername::class)];
+            $rules[$usernameAlt] = [
+                'required', 'string', 'min:'.Constants::MIN_USERNAME_CHARACTERS, 'max:'.Constants::MAX_USERNAME_CHARACTERS, 'unique:users', resolve(PoliteUsername::class), resolve(Username::class),
+            ];
+        }
+
+        if (Config::get('fortify.username')) {
+            $rules['email'] = ['email', 'required'];
```

Comment:
can you explain to me what this is needed for

Reply:
Hm, was to prevent some failure within tests but need to get back to it, I'll report back

Reply:
Yeah, for my latest test `should work with username authentication` I would get the following error without it ; 
`SQLSTATE[23000]: Integrity constraint violation: 19 NOT NULL constraint failed: users.email (SQL: insert into "users" ("name", "username", "password", "updated_at", "created_at") values (John Doe, alfonsobries, $2y$10$CG2TTFTJANAWZEagDOCuvOF.d95f3KehxQSvzuGISQrx6EmrOOjYG, 2021-03-26 09:20:39, 2021-03-26 09:20:39))`

Reply:
yes, but explain to me why the `$rules['email'] = ['email', 'required'];` line is needed. This is new in your PR while the existing implementation already works with `username` login (try on MSQ for example), so what constitutes this change?